Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/FileXmlElementAccess.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.dataformat.access;
+
+import java.math.BigInteger;
+import java.net.URI;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.kitodo.api.dataformat.MediaUnit;
+import org.kitodo.api.dataformat.MediaVariant;
+import org.kitodo.dataformat.metskitodo.DivType;
+import org.kitodo.dataformat.metskitodo.DivType.Fptr;
+import org.kitodo.dataformat.metskitodo.FileType;
+import org.kitodo.dataformat.metskitodo.Mets;
+
+public class FileXmlElementAccess {
+
+    /**
+     * The data object of this file XML element access.
+     */
+    private final MediaUnitMetsReferrerStorage mediaUnit;
+
+    /**
+     * Public constructor for a new media unit. This constructor can be used
+     * with the service loader to get a new instance of media unit.
+     */
+    public FileXmlElementAccess() {
+        mediaUnit = new MediaUnitMetsReferrerStorage();
+    }
+
+    /**
+     * Constructor for developing a media unit from a METS {@code <div>}
+     * element.
+     * 
+     * @param div
+     *            METS {@code <div>} element to be evaluated
+     * @param mets
+     *            the Mets structure is searched for corresponding uses
+     * @param useXmlAttributeAccess
+     *            list of media variants from which the media variant for the
+     *            given use is taken
+     */
+    FileXmlElementAccess(DivType div, Mets mets, Map<String, MediaVariant> useXmlAttributeAccess) {
+        this();
+        mediaUnit.setDivId(div.getID());
+        Map<MediaVariant, URI> mediaFiles = div.getFptr().parallelStream().map(Fptr::getFILEID)
```

Comment:
What is happening in this stream?

Reply:
[Already existing here](https://github.com/kitodo/kitodo-production/blob/master/Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MediaUnit.java#L81-L89)